odoo/models.py

```diff
@@ -5551,6 +5553,88 @@ def diff(self, other):
+            'low': 10,  # minimal representative set
+            'medium': 100,  # average database load
+            'high': 1000 # maxi database load
+        }
+
+    @property
+    def _populate_model_dependencies(self):
+        """
+            Auto compute dependencies of the model, aka models that should be populated before
+            this one base ond many2one fields. In case of ambiguity (cycle) the field is ignored.
+            This can be override to hardcode (partial) dependencies, especially to solve cycles.
+        """
+        deps = []
+        field_black_list = ('create_uid', 'write_uid')
+        def is_electible_dependency(field):
+            return field.type == 'many2one' and field.store and not field.compute and field.name not in field_black_list
+
+        processed = set()
+        init_model = None
+
+        # this has an awfull complexity since this will be used on all models.
+        # A nice solution would be to precompute all loop in model graph, and check if loops containing
+        # self._name also contains field.comodel_name
+        # in other words, store all field that participate to a cycle and and blacklist them
+
+        def check_loops(field):
+            model = self.env[field.comodel_name]
+            if model == init_model: # loop to initial model
+                raise ValueError('Cyclic dependency detected for %s' % model)
+            if model._name in processed: # already visited, stop iteration
+                return
+            processed.add(model._name)
+
+            for field in model._fields.values():
+                if not is_electible_dependency(field):
+                    continue
+                check_loops(field)
+
+        for field in self._fields.values():
+            if is_electible_dependency(field):
+                processed = set()
+                init_model = self
+                try:
+                    check_loops(field)
+                    deps.append(field.comodel_name)
+                except ValueError:
+                    pass
+        return deps
+
+    def _populate_database(self, scale):
+        # todo add info that model is populated somewhere
+        batch_size = 1000
+        scale_limit = self._populate_database_scales[scale]
+
+        record_count = 0
+        create_values = []
+        complete = False
+        field_generators = self._populate_database_fields_generators()
+        news = self.browse()
+        if not field_generators:
+            return news # maybe create an automatic generator?
+        generator = populate.chain_generators(field_generators, self._name)
+        while record_count <= scale_limit or not complete:
+            values, complete = next(generator)
+            create_values.append(values)
+            record_count += 1
+            if len(create_values) >= batch_size:
+                _logger.info('Batch: %s/%s', record_count, scale_limit)
+                news |= self.create(create_values)
```

Comment:
This is inefficient.  Append the recorset to a list instead, and then concatenate them at the end.
```python
return self.concat(*records_list)
```